packages/grafana-runtime/src/services/live.ts

```diff
@@ -31,6 +31,9 @@ export interface StreamingFrameOptions {
   maxLength: number; // 1000
   maxDelta: number; // how long to keep things
   action: StreamingFrameAction; // default will append
+
+  /** optionally format field names based on labels */
+  legendFormat?: string;
```

Comment:
nitpick, legendFormat naming feels a bit disconnected from the DataFrame structure 

```suggestion
  displayNameFormat?: string;
```